src/test/java/uk/gov/pay/ledger/report/dao/ReportDaoIT.java

```diff
@@ -378,4 +379,83 @@ public void shouldSumAmountWhenTotalAmountIsNull() {
         assertThat(refundsStatistics.getCount(), is(1L));
         assertThat(refundsStatistics.getGrossAmount(), is(1000L));
     }
+
+    @Test
+    public void shouldGetTransactionVolumesGroupedByTimeseries() {
+        aTransactionFixture()
+                .withGatewayAccountId("100")
+                .withTransactionType(TransactionType.PAYMENT.name())
+                .withCreatedDate(ZonedDateTime.parse("2019-09-30T00:00:00.100Z"))
+                .withAmount(1000L)
+                .withState(TransactionState.SUCCESS)
+                .withLive(true)
+                .insert(rule.getJdbi())
+                .toEntity();
+
+        aTransactionFixture()
+                .withGatewayAccountId("100")
+                .withTransactionType(TransactionType.PAYMENT.name())
+                .withCreatedDate(ZonedDateTime.parse("2019-09-30T08:30:00.000Z"))
+                .withAmount(1000L)
+                .withTotalAmount(1500L)
+                .withState(TransactionState.SUCCESS)
+                .withLive(true)
+                .insert(rule.getJdbi())
+                .toEntity();
+
+        aTransactionFixture()
+                .withGatewayAccountId("100")
+                .withTransactionType(TransactionType.PAYMENT.name())
+                .withCreatedDate(ZonedDateTime.parse("2019-09-30T08:40:00.000Z"))
+                .withAmount(1000L)
+                .withTotalAmount(1500L)
+                .withState(TransactionState.SUBMITTED)
+                .withLive(true)
+                .insert(rule.getJdbi())
+                .toEntity();
+
+        aTransactionFixture()
+                .withGatewayAccountId("100")
+                .withTransactionType(TransactionType.PAYMENT.name())
+                .withCreatedDate(ZonedDateTime.parse("2019-09-30T09:20:00.000Z"))
+                .withAmount(1000L)
+                .withState(TransactionState.SUCCESS)
+                .withLive(true)
+                .insert(rule.getJdbi())
+                .toEntity();
+
+        aTransactionFixture()
+                .withGatewayAccountId("100")
+                .withTransactionType(TransactionType.PAYMENT.name())
+                .withCreatedDate(ZonedDateTime.parse("2019-09-30T18:20:00.000Z"))
+                .withAmount(1000L)
+                .withState(TransactionState.ERROR)
+                .withLive(true)
+                .insert(rule.getJdbi())
+                .toEntity();
+
+        List<TimeseriesReportSlice> timeseriesReportSlices = reportDao.getTransactionsVolumeByTimeseries(
+                ZonedDateTime.parse("2019-09-30T00:00:00.000Z"),
+                ZonedDateTime.parse("2019-09-30T23:59:59.999Z")
+        );
+        assertThat(timeseriesReportSlices.size(), is(4));
+
+        assertThat(timeseriesReportSlices.get(0).getTimestamp().getHour(), is(0));
+        assertThat(timeseriesReportSlices.get(0).getAllPayments(), is(1));
```

Comment:
do we not care about amount, total amount, completed payments and errored payments here?

Reply:
I don't think so -- each group is checking a different scenario/ angle without asserting on every single value. So it's assuming that completed/ errored payments are covered somewhere else. Hopefully that's the case 👀

Reply:
Hmm maybe I don't really understand what you're trying to say, but from a java objects point of view won't be there be values for the fields in the `timeseriesReportSlices.get(0)` object? Here the completed payments field has the value of 1, do we not care about it?

Reply:
Yeah poorly explained by me! Just trying to strike a balance between covering the functionality of the query and not being overly verbose or restrictive to avoid flake. I'll add some extra assertions to cover what you've mentioned